File: c-sharp/FizzBuzz/Implementation.cs
namespace CSharp.FizzBuzz;

public class FizzBuzz
{
    private readonly int start;
    private readonly int end;

    public FizzBuzz(
        int start,
        int end)
    {
        this.start = start;
        this.end = end;
    }

    public IEnumerable<string> GetResult()
    {
        for (var i = this.start; i <= this.end; i++)
        {
            if (i % 7 == 0)
            {
                yield return $"{i}";
            }
            else if (i % 5 == 0 && i % 3 == 0)
            {
                yield return "FizzBuzz";
            }
            else if (i % 3 == 0)
            {
                yield return "Fizz";
            }
            else if (i % 5 == 0)
            {
                yield return "Buzz";
            }
            else
            {
                yield return $"{i}";
            }
        }
    }
}

File: f-sharp/FizzBuzz/Implementation.fs
module FSharp.FizzBuzz

let fizzBuzz start limit : string list =
    let (|Fizz|Not|) input = if input % 3 = 0 then Fizz else Not
    let (|Buzz|Not|) input = if input % 5 = 0 then Buzz else Not
    let (|By7|NotBy7|) input = if input % 7 = 0 then By7 else NotBy7

    [ for i in start..limit do
          match i with
          | Fizz & Buzz & NotBy7 -> "FizzBuzz"
          | Fizz & NotBy7 -> "Fizz"
          | Buzz & NotBy7 -> "Buzz"
          | number -> number.ToString() ]
